GameServerLib/GameObjects/Other/CollisionHandler.cs

```diff
@@ -126,14 +128,18 @@ public void Update()
                         obj.OnCollision(obj2);
                     }
 
-                    if (!IsCollisionObject(obj2))
-                    {
-                        continue;
-                    }
-
                     // TODO: Implement repathing if our position within the next few ticks intersects with another GameObject (assuming we are moving; !IsPathEnded).
                 }
             }
         }
+
+        /// <summary>
+        /// Used to reinitialize a QuadTree's sectors when objects may have moved out of sectors, which makes them unable to be removed.
+        /// </summary>
+        private void UpdateQuadTree()
+        {
+            _quadDynamic = new QuadTree<IGameObject>(_quadDynamic.MainRect.Width, _quadDynamic.MainRect.Height, _objectBounds);
```

Comment:
There will be a problem here because the starting point shouldn't be (0,0) but instead some negative value of the NavGrid.